### PR TITLE
Finalize initial list of components for KFP and Airflow

### LINKED ISSUE #1698: Identify the initial list of Kubeflow pipeline components to be pre-installed with Elyra
**Describe the issue**
Similar to what we do with Elyra runtime images, we should identify a subset of runtime-specific components to be pre-installed with Elyra so that we have a usable development environment right after installation.

### LINKED ISSUE #1728: Identify the initial list of Apache Airflow pipeline components to be pre-installed with Elyra
**Describe the issue**
Similar to what we do with Elyra runtime images, we should identify a subset of Airflow operators to be pre-installed with Elyra so that we have a usable development environment right after installation.

Related to #1698, which is the same issue but for KFP components

### PR DESCRIPTION
**Pull request by kiersten-stokes**
### What changes were proposed in this pull request?
Resolves #1728 and resolves #1698 by providing the initial list of components for Airflow and KFP. Also adds simple error messages to cases where the component catalog cannot be accessed, as suggested in #1766.

### How was this pull request tested?
See screenshots below for error-checking examples.

Malformed JSON:
Screen Shot 2021-06-17 at 1 37 51 PM

Component catalog FNFE:
Screen Shot 2021-06-17 at 1 39 09 PM

Fixes #1728 
Fixes #1698 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: elyra/pipeline/registry.py [hunk 1]
```diff
@@ -161,11 +161,18 @@ def get_common_config(self, config_name):
...
+            # then sys.prefix, where installed files will reside (factory data)
+            catalog_dir = os.path.join(jupyter_core.paths.ENV_JUPYTER_PATH[0], self._components_dir)
+            catalog_file = os.path.join(catalog_dir, f"{self._type}_component_catalog.json")
+            with open(catalog_file, 'r') as f:
+                catalog_json = json.load(f)
+        except FileNotFoundError:
+            self.log.error(f"The component catalog could not be found at path '{catalog_file}'.")
+            raise FileNotFoundError(f"The component catalog could not be found at path '{catalog_file}'.")
+        except Exception:
+            self.log.error(f"The component catalog at path '{catalog_file}' could not be loaded.")
```
Comment (ptitzler) [inline elyra/pipeline/registry.py:174]: should log exception text to enable troubleshooting

Comment (kiersten-stokes) [reply elyra/pipeline/registry.py:174]: Good point, thank you!

File: elyra/pipeline/registry.py [hunk 2]
```diff
@@ -721,7 +728,8 @@ def get_properties(self, processor_type, component_id):
             # Find component with given id in component catalog
             component = parser.return_component_if_exists(component_id)
             if component is None:
-                raise ValueError(f"Component with ID {component_id} not found.")
+                self.log.error(f"Component with ID '{component_id}' could not be found in the component catalog.")
```
Comment (ptitzler) [inline elyra/pipeline/registry.py:731]: the message text should identify the catalog type to assist with troubleshooting

**Comment by akchinSTC** [comment]
@lresende - our list of components for kfp is dwindling, the filter and grep component has little application according to ptitzler so if thats I feel like we should take a look at some other ones to add ..perhaps the kfserving and katib ones from tommy? or do a little more digging on some contributor repositories for some more basic components to add?

**Comment by akchinSTC** [comment]
Looking to add the kfserving one to the list right now. 
@ptitzler - since the filter component is not particularly useful, and only works as a single node, should we remove it?

**Comment by ptitzler** [comment]
> since the filter component is not particularly useful, and only works as a single node, should we remove it?

For the time being it's probably okay to leave in because it is somewhat useful for _illustrative purposes_ because it  does not require any third-party prerequisites.

**Review by akchinSTC — APPROVED** [review]

**Comment by akchinSTC** [comment]
Moved error checking changes into PR https://github.com/elyra-ai/elyra/pull/1811 and added kf-serving component

**Pull request merged**